lib/std/os.zig

```diff
@@ -2542,6 +2542,17 @@ pub fn fstat(fd: fd_t) FStatError!Stat {
         }
     }
 
+    if (std.Target.current.os.tag == .netbsd) {
+        switch (errno(system.__fstat50(fd, &stat))) {
```

Comment:
lets share the error code paths with the generic posix ones below?

```zig
switch (errno(switch(std.Target.current.os.tag) {
     .netbsd => system.__fstat50(fd, &stat),
     else => system.fstat(fd, &stat),
})) {
    ....
}
```

Reply:
I was thinking of something like this to avoid excessive nesting:

```zig
const fstat_symbol = switch (std.Target.current.os.tag) {
     .netbsd => "__fstat50",
     else => "fstat",
};
switch (errno(@field(system, fstat_symbol))) {
 ...
}
```